server/athenian/api/internal/miners/github/deployment.py

```diff
@@ -2602,3 +2611,149 @@ async def load_jira_issues_for_deployments(
                 prs[pri] = np.sort(np.array(release_keys, dtype="S"))
 
     return issues
+
+
+async def hide_outlier_first_deployments(
+    deployment_facts: pd.DataFrame,
+    account: int,
+    meta_ids: Sequence[int],
+    mdb: Database,
+    pdb: Database,
+    threshold: float = 100.0,
+) -> None:
+    """Hide the outlier first deployments by deleting their facts from pdb.
+
+    A deployment is an outlier first for a repo if:
+    - it's the first deployment for a given repo and environment
+    - the time distance with the repository creation is at least `threshold` times
+      the median of time distances of each deployment with the previous one
+    """
+    log = logging.getLogger(f"{__name__}.hide_outlier_first_deployments")
+    outlier_deploys = await _search_outlier_first_deployments(
+        deployment_facts, meta_ids, mdb, threshold,
+    )
+
+    if not outlier_deploys:
+        return
+
+    log.info("Clearing %d outlier first deployments", len(outlier_deploys))
+
+    stmts: list[Executable] = []
+
+    # group outlier deploys by name
+    grouped_deploys = [
+        (name, [d.repository_full_name for d in name_groups])
+        for name, name_groups in groupby(
+            sorted(outlier_deploys, key=attrgetter("deployment_name")),
+            key=attrgetter("deployment_name"),
+        )
+    ]
+
+    tables = (GitHubCommitDeployment, GitHubPullRequestDeployment, GitHubReleaseDeployment)
+    for Table in tables:
+        where_clause = sa.and_(
+            Table.acc_id == account,
+            sa.or_(
+                *[
+                    sa.and_(Table.deployment_name == g[0], Table.repository_full_name.in_(g[1]))
+                    for g in grouped_deploys
+                ],
+            ),
+        )
+        stmts.append(sa.delete(Table).where(where_clause))
```

Comment:
If it were a SELECT, I would suggest a UNION ALL, but we delete here, so the trick won't work. However, leaving as-is means a linear scan over all the records belonging to the account. Let's execute independent `sa.delete(Table)` for each group

Reply:
done